### PR TITLE
Support specified shard number when create collection

### LINKED ISSUE #6690: Support specified shard number when create collection.
#### Is your feature request related to a problem? Please describe:
currently Milvus didn't support user specified shard nums, which brings a  lot of confusing on why the cluster is not balanced.

#### Describe the solution you'd like:
User should be able to specify a shard number when create collection. Valid shard number should be a number between 1-4096, thus milvus can be balanced under 1000 nodes cluster.

#### Additional context:
Partition * Shard would be a very big number, thus growing segment could be a fairly large number. 
If a user has 64 shard with 1024 partitions, growing segment numbers should be 65536 and this consumes huge memories in both datanode and querynode, any suggestions on this?

### PR DESCRIPTION
**Pull request by czs007**
Resolves: #6690 
Signed-off-by: zhenshan.cao <zhenshan.cao@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/task.go [hunk 1]
```diff
@@ -93,12 +93,12 @@ func (t *CreateCollectionReqTask) Execute(ctx context.Context) error {
 	if t.Req.CollectionName != schema.Name {
 		return fmt.Errorf("collection name = %s, schema.Name=%s", t.Req.CollectionName, schema.Name)
 	}
-
 	if t.Req.ShardsNum <= 0 {
 		log.Debug("Set ShardsNum to default", zap.String("collection name", t.Req.CollectionName),
 			zap.Int32("defaultShardsNum", defaultShardsNum))
 		t.Req.ShardsNum = defaultShardsNum
 	}
+	log.Debug("ShardsNum is ", zap.Any("shardnum", t.Req.ShardsNum))
```
Comment (xiaofan-luan) [inline internal/rootcoord/task.go:101]: You should at least print collection name here

**Review by xiaofan-luan — CHANGES_REQUESTED** [review]
We need test on create collection, then describe and check if shard information is there.

**Comment by xiaofan-luan** [comment]
need more unit test here to prove this patch will work, others lgtm

**Comment by czs007** [comment]
@xiaofan-luan  PTAL

**Review by xiaofan-luan — APPROVED** [review]

**Pull request merged**